packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -443,9 +443,13 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox {
 
   void _handleDragEnd(DragEndDetails details) {
     if (_position.value >= 0.5)
-      _positionController.forward();
+      _positionController.forward().whenComplete(() {
```

Comment:
That happens only when the switch is not completely slid.
Should i do something like 
```dart
....whenComplete(() {
        WidgetsBinding.instance.addPostFrameCallback((_) {
          if (value && attached) positionController.forward();
        });
      });
```

Reply:
Yeah it looks like delaying it gets rid of the unnecessary steps. But I'm slightly in favor of adding a new boolean state (in addition to `_value`), for the "real" value the switch currently represents, mainly because with the `whenComplete` the code will still be a bit hard to reason about without extensive comments, and adding the missing state sounds like a more "proper" fix to me. What do you think?

Reply:
I don't think we need a new variable, we can get the real value(visual appearance) using `positionController.isDismissed` and `positionController.isCompleted`. How can we use that to solve this issue.

Reply:
The problem as I see it is that currently when the user drags the thumb, the `SwitchState` sees no state change, even if the thumb has visually moved to the other end. As a result the render object is not updated despite the `value` and the visual appearance are now inconsistent. If there's a new state variable to represent the visual appearance then we would be able to tell that we need to update the render object of the switch. I guess the variable will represent something slightly different from `positionController.isDismissed` or `positionController.isCompleted`, because we don't want to reset the switch while the user is still interacting with it.